src/HealthGPS/static_linear_model.cpp

```diff
@@ -0,0 +1,118 @@
+#include "static_linear_model.h"
+#include "runtime_context.h"
+
+#include "HealthGPS.Core/exception.h"
+
+namespace hgps {
+
+StaticLinearModel::StaticLinearModel(std::vector<core::Identifier> risk_factor_names,
+                                     LinearModelParams risk_factor_models,
+                                     Eigen::MatrixXd risk_factor_cholesky)
+    : risk_factor_names_{std::move(risk_factor_names)},
+      risk_factor_models_{std::move(risk_factor_models)},
+      risk_factor_cholesky_{std::move(risk_factor_cholesky)} {
+
+    if (risk_factor_names_.empty()) {
+        throw core::HgpsException("Risk factor names list is empty");
+    }
+    if (risk_factor_models_.intercepts.empty() || risk_factor_models_.coefficients.empty()) {
+        throw core::HgpsException("Risk factor models mapping is incomplete");
+    }
+    if (!risk_factor_cholesky_.allFinite()) {
+        throw core::HgpsException("Risk factor Cholesky matrix contains non-finite values");
+    }
+}
+
+HierarchicalModelType StaticLinearModel::type() const noexcept {
+    return HierarchicalModelType::Static;
+}
+
+std::string StaticLinearModel::name() const noexcept { return "Static"; }
+
+void StaticLinearModel::generate_risk_factors(RuntimeContext &context) {
+
+    for (auto &person : context.population()) {
+
+        // Approximate risk factor values with linear models.
+        linear_approximation(person);
+
+        // Correlated residual sampling.
+        auto samples = correlated_samples(context);
+    }
+}
+
+void StaticLinearModel::update_risk_factors(RuntimeContext &context) {
+
+    for (auto &person : context.population()) {
+
+        // Only newborns should be initialised.
+        if (person.age > 0) {
+            continue;
+        }
+
+        // Approximate risk factor values with linear models.
+        linear_approximation(person);
+
+        // Correlated residual sampling.
+        auto samples = correlated_samples(context);
+    }
+}
+
+Eigen::VectorXd StaticLinearModel::correlated_samples(RuntimeContext &context) {
+
+    // Correlated samples using Cholesky decomposition.
+    Eigen::VectorXd samples{risk_factor_names_.size()};
+    for (size_t i = 0; i < risk_factor_names_.size(); i++) {
+        samples[i] = context.random().next_normal(0.0, 1.0);
+    }
```

Comment:
If you `#include <ranges>` this can be written as:

```suggestion
    std::ranges::generate(samples, [&context] { return context.random().next_normal(0.0, 1.0); });
```